src/octoprint/server/api/settings.py

```diff
@@ -964,6 +965,11 @@ def _saveSettings(data):
                 ["serial", "ignoreEmptyPorts"], data["serial"]["ignoreEmptyPorts"]
             )
 
+        if "encodingScheme" in data["serial"]:
+            value = data["serial"]["encodingScheme"]
+            if value in ("ascii", "latin_1"):
```

Comment:
tbh, I would actually go so far here to just check if it's a valid encoding identifier in general, and if so use it. Don't see why we should limit it explicitly to ascii and *latin1* of all things. If this is made configurable, I'd suggest to make it *really* configurable. If someone wants to run this against a device that talks `cp1252` for some reason, who am I to judge?